### PR TITLE
feat: change request_id to disallow null and update index

### PR DESCRIPTION
**Pull request by pavelbrm**
### Summary

This PR updates the `time_limited_v2_order_creds_unique` constraint on the `time_limited_v2_order_creds` table, thus allowing credentials for multiple unique batches within the same period, which is necessary for Multi Device Refresh.

### Type of Change

- [x] Product feature
- [x] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before Requesting Review

- [x] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [x] Is there a clear title that explains what the PR does?
- [x] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security and/or privacy review if needed
- [x] Have you performed a self review of this PR?

### CODE REVIEW

File: migrations/0067_tlv2_request_id_not_null.up.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE time_limited_v2_order_creds ALTER COLUMN request_id SET NOT NULL;
```
Comment (pavelbrm) [inline migrations/0067_tlv2_request_id_not_null.up.sql:1]: At the moment of writing, there have been no rows with `request_id` `NULL` in the table (both Staging and Production).

File: services/skus/datastore_test.go [hunk 2]
```diff
@@ -498,7 +502,7 @@ func createOrderAndIssuer(t *testing.T, ctx context.Context, storage Datastore,
 
 	repo := repository.NewIssuer()
 	issuer, err := repo.Create(ctx, storage.RawDB(), model.IssuerNew{
-		MerchantID: test.RandomString(),
+		MerchantID: model.MerchID,
```
Comment (pavelbrm) [inline services/skus/datastore_test.go:505]: There is only one merchant, and that is us, `brave.com`. This randomisation was totally useless.

File: services/skus/skustest/skustest.go [hunk 3]
```diff
@@ -7,60 +7,46 @@ import (
...
 
-	version, dirty, _ := migrate.Version()
-	if dirty {
-		assert.NoError(t, migrate.Force(int(version)))
-	}
+	m, err := pg.NewMigrate()
+	must.NoError(t, err)
 
-	if version > 0 {
-		assert.NoError(t, migrate.Down())
```
Comment (pavelbrm) [inline services/skus/skustest/skustest.go:35]: No need to down every time, just once is enough (in a separate test).

**Review by DJAndries — APPROVED** [review]

**Review by kdenhartog — APPROVED** [review]
LGTM

**Review by evq — APPROVED** [review]
lgtm! there is a race that this PR in isolation would expose us to - however the plan as I understand it is to deploy this alongside https://github.com/brave-intl/bat-go/pull/2522 where it is fixed

**Comment by pavelbrm** [comment]
> lgtm! there is a race that this PR in isolation would expose us to - however the plan as I understand it is to deploy this alongside #2522 where it is fixed

Yes, this and #2522, as well as other related PRs are all planned to go in one release, therefore the risk of the race is minimal if not zero.

**Pull request merged**